Fix MariaDB database detection

amottier opened the pull request:
So far detection of database product was relying on `HibernateStore` code that report MariaDB as being a MySQL database (to avoid code duplication are both databases are still quite similar).

In order to give a more accurate information about detected database product to the admin this commit duplicate and tune some code of `HibernateStore` and `DatabaseProduct` to correctly report MariaDB database product.

tmortagne commented:
It's not very clear to me why we absolutely want to filter the short database name (like "MySQL"), why not directly showing the unfiltered `hibernateStore.getDatabaseMetaData().getDatabaseProductName()` instead ?

amottier commented:
> It's not very clear to me why we absolutely want to filter the short database name (like "MySQL"), why not directly showing the unfiltered `hibernateStore.getDatabaseMetaData().getDatabaseProductName()` instead ?

Actually I didn't realize that `HibernateStore` expose `getDatabaseMetaData` so I used `getDatabaseProductName`. After your comment I reworked my code in order to use `getDatabaseMetaData` and replaced my initial commit.

It should be ready for another review.

Thanks for the suggestion @tmortagne  :+1:
